src/Acuminator/Acuminator.Tests/Verification/VerificationHelper.cs

```diff
@@ -183,5 +195,49 @@ public static async Task<IEnumerable<Diagnostic>> GetCompilerDiagnosticsAsync(Do
 			var semanticModel = await document.GetSemanticModelAsync().ConfigureAwait(false);
 			return semanticModel.GetDiagnostics();
 		}
+
+		/// <summary>
+		/// Compile internal source code into memory array 
+		/// </summary>
+		/// <param name="sourceCodes">Source codes of library</param>
+		/// <returns>Byte array from compiled binary dll assembly</returns>
+		private static byte[] BuildAssemblyFromSources(string[] sourceCodes)
+		{
+			string assemblyName = Path.GetRandomFileName();
+
+			CSharpCompilation compilation = CSharpCompilation.Create(
+					assemblyName,
+					syntaxTrees: sourceCodes.Select(code => CSharpSyntaxTree.ParseText(text: code)),
+					references: new MetadataReference[] { CorlibReference,
+														  SystemCoreReference,
+														  CSharpSymbolsReference,
+														  CodeAnalysisReference,
+														  PXDataReference,
+														  PXCommonReference },
+					options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+			// Emit the image of this assembly 
+			byte[] image = null;
+			using (var ms = new MemoryStream())
+			{
+				var emitResult = compilation.Emit(ms);
+				if (!emitResult.Success)
+				{
+					List<Exception> failtureMessages = new List<Exception>();
+					IEnumerable<Diagnostic> failures = emitResult.Diagnostics.Where(diagnostic =>
+								   diagnostic.IsWarningAsError ||
+								   diagnostic.Severity == DiagnosticSeverity.Error);
+
+					foreach (Diagnostic diagnostic in failures)
+					{
+						failtureMessages.Add(new ArgumentException(string.Format("{0}: {1}", diagnostic.Id, diagnostic.GetMessage())));
+					}
+
+					throw new ArgumentException(BuildFailMessage, new AggregateException(failtureMessages));
```

Comment:
`AggregateException` must be used only for aggregating several other exceptions.
In this case either wrap each error message into a separate exception or concat them using a string builder / `String.Join`.